app/internal/page/recover.go

```diff
@@ -0,0 +1,46 @@
+package page
+
+import (
+	"fmt"
+	"net/http"
+	"runtime/debug"
+	"strings"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/localize"
+)
+
+type recoverError struct {
+	err   any
+	stack []byte
+}
+
+func (e recoverError) Error() string { return "recover error" }
+func (e recoverError) Title() string { return fmt.Sprint(e.err) }
+func (e recoverError) Data() any     { return string(e.stack) }
+
+func Recover(tmpl template.Template, logger Logger, bundle localize.Bundle, next http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if err := recover(); err != nil {
+				logger.Request(r, recoverError{err: err, stack: debug.Stack()})
+				w.WriteHeader(http.StatusInternalServerError)
+
+				appData := AppData{CookieConsentSet: true, Paths: Paths}
```

Comment:
This is the minimum I think to produce an OK looking page. It won't be perfect, but given we are producing high quality 😉 code there shouldn't be much chance of anyone ever seeing it.